Branch-SDK/Branch-SDK/Networking/BNCNetworkService.m

```diff
@@ -0,0 +1,279 @@
+//
+//  BNCNetworkService.m
+//  Branch-SDK
+//
+//  Created by Edward Smith on 5/30/17.
+//  Copyright © 2017 Branch Metrics. All rights reserved.
+//
+
+#import "BNCNetworkService.h"
+#import "BNCEncodingUtils.h"
+#import "BNCLog.h"
+#import "BNCDebug.h"
+
+#pragma mark BNCNetworkOperation
+
+@interface BNCNetworkOperation ()
+@property (copy)   NSMutableURLRequest*request;
+@property (copy)   NSHTTPURLResponse  *response;
+@property (strong) NSData             *responseData;
+@property (copy)   NSError            *error;
+@property (copy)   NSDate             *dateStart;
+@property (copy)   NSDate             *dateFinish;
+
+@property (strong) BNCNetworkService  *networkService;
+@property (strong) NSURLSessionTask   *sessionTask;
+@property (copy) void (^completionBlock)(BNCNetworkOperation*operation);
+@end
+
+#pragma mark - BNCNetworkService
+
+@interface BNCNetworkService () <NSURLSessionDelegate, NSURLSessionTaskDelegate> {
+    NSURLSession    *_session;
+    NSTimeInterval  _defaultTimeoutInterval;
+    NSInteger       _maximumConcurrentOperations;
+}
+
+- (void) startOperation:(BNCNetworkOperation*)operation;
+
+@property (strong, readonly) NSURLSession *session;
+@property (strong) NSOperationQueue *sessionQueue;
+@end
+
+#pragma mark - BNCNetworkOperation
+
+@implementation BNCNetworkOperation
+
+- (void) start {
+    [self.networkService startOperation:self];
+}
+
+- (void) cancel {
+    [self.sessionTask cancel];
+}
+
+- (NSString*) stringFromResponseData {
+    NSString *string = nil;
+    if ([self.responseData isKindOfClass:[NSData class]]) {
+        string = [[NSString alloc] initWithData:(NSData*)self.responseData encoding:NSUTF8StringEncoding];
+    }
+    if (!string && [self.responseData isKindOfClass:[NSData class]]) {
+        string = [NSString stringWithFormat:@"<NSData of length %ld.>",
+            (long)[(NSData*)self.responseData length]];
+    }
+    if (!string) {
+        string = self.responseData.description;
+    }
+    return string;
+}
+
+@end
+
+#pragma mark - BNCNetworkService
+
+@implementation BNCNetworkService
+
++ (id<BNCNetworkServiceProtocol>) new {
+    return [[self alloc] init];
+}
+
+- (instancetype) init {
+    self = [super init];
+    if (!self) return self;
+    _defaultTimeoutInterval = 60.0;
```

Comment:
The network timeout is set in the BNCServerInterface class.  This is a backup value because I needed some sort of sane value if something wasn't set correctly.